pkg/process/metadata/parser/java/yaml.go

```diff
@@ -0,0 +1,51 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package javaparser
+
+import (
+	"io"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+)
+
+type flattener struct {
+	props map[string]string
+}
+
+func (f *flattener) visit(path []string, node *yaml.Node) {
+	for i := 0; i < len(node.Content); i += 2 { // should be safer
+		current := node.Content[i]
+		next := node.Content[i+1]
```

Comment:
What gives us the guarantee that `len(node.Content)` is even?

Reply:
i put an early return for this